File: Placement Sheet/LinkedList/CheckPalindromeLinkedlist.java
import java.util.Stack;

public class CheckPalindromeLinkedlist {
    public static class Node {
        int data;
        Node next;

        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }

    private Node head = null;
    private int size = 0;

    // Insert...
    public void insert(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            size++;
            return;
        }
        Node currNode = head;
        while (currNode.next != null) {
            currNode = currNode.next;
        }
        currNode.next = newNode;
        size++;
    }

    // Display...
    public void display() {
        if (head == null) {
            System.out.println("List is empty...");
            return;
        }
        Node currNode = head;
        while (currNode != null) {
            System.out.print(currNode.data + " --> ");
            currNode = currNode.next;
        }
        System.out.println("Null");
    }

    public boolean isPalindrome() {
        if (head == null) {
            System.out.println("List is empty...");
            return false;
        }
        int[] values = new int[size];
        Node currNode = head;
        int index = 0;
        while (currNode != null) {
            values[index] = currNode.data;
            index++;
            currNode = currNode.next;
        }

        int startPtr = 0;
        int endPtr = values.length - 1;

        while (startPtr < endPtr) {
            if (values[startPtr] != values[endPtr]) {
                return false;
            }
            startPtr++;
            endPtr--;
        }
        return true;
    }

    public static void main(String[] args) {
        CheckPalindromeLinkedlist list = new CheckPalindromeLinkedlist();
        list.insert(1);
        list.insert(2);
        list.insert(3);
        list.insert(3);
        list.insert(2);
        list.insert(1);
        list.display();
        System.out.println("The list is palindrome: "+ list.isPalindrome());
    }
}

File: Placement Sheet/LinkedList/CheckPalindromeLinkedlist.ts
class ListNode {
    private data: number;
    private next: ListNode | null;

    constructor(data: number) {
        this.data = data;
        this.next = null;
    }

    getData(): number {
        return this.data;
    }

    getNext(): ListNode | null {
        return this.next;
    }

    setNext(node: ListNode): void {
        this.next = node;
    }
}

class LinkedList {
    private head: ListNode | null = null;

    //Append data...
    append(data: number): void {
        let newNode: ListNode | null = new ListNode(data);
        if (this.head == null) {
            this.head = newNode;
            return;
        }
        let currNode: ListNode | null = this.head;
        while (currNode.getNext() !== null) {
            currNode = currNode?.getNext()!;
        }
        currNode.setNext(newNode);
    }

    //Display...
    display(): void {
        if (this.head == null) {
            console.log("List is empty...");
            return;
        }
        let currNode: ListNode | null = this.head;
        let result: string = "";
        while (currNode !== null) {
            result += currNode.getData() + " --> ";
            currNode = currNode.getNext();
        }
        result += "Null";
        console.log(result);
    }

    //Check Palindrome...
    isPalindrome(): boolean {
        if (this.head == null || this.head.getNext() == null) {
            console.log("List is empty...");
            return false;
        }
        let values: number[] = [];
        let currNode: ListNode | null = this.head;
        while (currNode !== null) {
            values.push(currNode.getData());
            currNode = currNode.getNext();
        }
        let startPtr: number = 0;
        let endPtr: number = values.length - 1;

        while (startPtr < endPtr) {
            if (values[startPtr] !== values[endPtr]) {
                return false;
            }
            startPtr++;
            endPtr--;
        }
        return true;
    }
}

let list: LinkedList = new LinkedList();
list.append(1);
list.append(2);
list.append(3);
list.append(3);
list.append(2);
list.append(1);
console.log("LinkedList: ");
list.display();
console.log(`The List is palindrome: ${list.isPalindrome()}`);

export { };